react/src/components/atoms/forms/InputTextFuzzy/index.js

```diff
@@ -42,17 +50,23 @@ class InputTextFuzzy extends React.Component {
     return suggestions;
   }
   handleChange = (e) => {
+    e.persist();
     const { value } = e.target;
-    if (is.fn(this.props.onChange)) {
-      this.props.onChange(e, { value });
-    }
     if (is.empty(value)) {
       if (this.props.renderDefaultSuggestion) {
         const suggestions = this.optionsToSuggestions(this.props.options);
         this.setState({
           suggestions,
           value: ''
+        }, () => {
+          if (typeof this.props.onChange === 'function') {
+            this.props.onChange({ event: e, value, suggestions });
+          }
         });
+      } else {
```

Comment:
is this change on purpose? moving the callback into setState in the if block will only callback if value is empty?

Reply:
since we are not using state in props.onChange function, what's the difference putting it in callback vs outside of setState? if we pass it as callback we will have to do it in the setState below also